AjobK/Athena
# 240 dockerize project

Closes #240 

<br />

 **What is it supposed to do**  
The branch adds support for running this project in (three) docker containers via docker-compose by providing configuration files for docker.

<br />

 **How can we test your branch**  
The functionality can be tested by running the project in docker on your local machine. For this, docker needs to be installed (check the wiki on how to do that). When docker is installed, fill in the ormconfig.docker.json file (see wiki), go to the Seaqull folder in your terminal, and execute ```docker compose up --build```. When it is done, go to http://localhost in your browser, you should see the application running, everything should work as it does when running the code.

<br />

**1. General checklist**
- [ ] b. Tested unhappy flow.  _Not relevant_
- [ ] d. No code with high complexity. (Big O) _Not relevant_
- [ ] h. Lines no longer than 'guideline' in IDE. _Not relevant_
- [ ] k. (JS) Curly braces on the sameline as reserved keywords *(`if () {` instead of `{` on line below)*. _Not relevant_
- [ ] l. (JS) No double quotes. _Not applicable_
- [ ] m. (JS) Single newline above unique code structures and reserved keywords. _Not relevant_
- [ ] n. (JS) Variables bundled where possible. _Not relevant_
- [ ] o. (JS) No semi-colons except where they have a functional necessity, such as for-loops. _Not relevant_
- [ ] p. (JS) No code that is meant exclusively for development, such as console.logs for data preview. _Not relevant_

<br />

**2. If relevant, front-end checklist**  _Not relevant_

<br />

**3. If relevant, back-end checklist** 
- [ ] a. No unhandled promise rejections. _Not relevant_
- [ ] c. No primary-key violations. _Not relevant_
- [ ] d. Normalization, atleast [3NF](https://en.wikipedia.org/wiki/Database_normalization). _Not relevant_
- [ ] e. Used guards (middleware) where necessary and tested if route is guarded. _Not relevant_

<br />

**4. If relevant, test these browsers** _Not relevant_

---

**PR Rules**  
*1. Add relevant labels.*  
*2. Select atleast two reviewers.*  
*3. Assign all reviewers allowed to merge the branch, including yourself.*  
*4. If relevant to the sprint, put related issue in 'Awaiting Approval' column on project board.*  
*5. Add to the corresponding [milestone](https://github.com/AjobK/Seaqull/milestones).*

<br />

**Reviewer info**  
*1. You can use aliases to refer to specific PR checklist items. For example 'Hey @JohnDoe, for this line please check 3.a and 1.d.'*  
*2. Where possible, please use the suggestions feature in github so the author has a better idea of the solution you had in mind.*   

<br />

> This form should be filled in by the author of this PR.
